simhash/__init__.py

```diff
@@ -105,20 +107,12 @@ def build_by_features(self, features):
                 w = f[1]
             for i in range(self.f):
                 v[i] += w if h & masks[i] else -w
-        ans = 0
-        for i in range(self.f):
-            if v[i] > 0:
-                ans |= masks[i]
-        self.value = ans
+        binary_str = ''.join(['0' if i <= 0 else '1' for i in v])
+        self.value = int(binary_str, 2)
 
     def distance(self, another):
         assert self.f == another.f
-        x = (self.value ^ another.value) & ((1 << self.f) - 1)
-        ans = 0
-        while x:
-            ans += 1
-            x &= x - 1
-        return ans
+        return popcount(self.value ^ another.value)
```

Comment:
Hi, thanks for the pull request.
I'm trying to understand the `popcount` here. 
Have you done some performance test to show that `popcount` is more efficient?
Or maybe you have the source code of `popcount`?
I'm trying to find it out, but not found anything about its implementation yet.
:-)

Reply:
`popcount` is used to get the number of `1` in binary.

`gmpy2` is using `GMP` libary's `popcount` function. There is also a cpu instruction named `popcount`, I guess `GMP` should be able to call it. I cpu instruction should be much faster than iterate the bit array to find the `1`s.

I've show the performance test in #43, it's 10x faster.

Reply:
Thanks!, that helps a lot and it makes sense now. Just curious why the 2 methods produced different results because the old method is to get the number of 1 in binary too. 🙂

Reply:
By the way, one more thing, in case cpu doesn't have that instruction popcount, can it fall back to less efficient ways to get the same result? 🙂

Reply:
The `distance` function produce same result `19`.
But the two version produce different value, `864692470817131398` and `7048486158682030128` respectively. Because new method treat the first bit as most significant bit. For example, when `v` is `001`, the original method produces `4`, but the new method produces `1`.

Reply:
Don't worry, `GMP` library should always return the right result regardless of which cpu is using.